### PR TITLE
JAMES-2394 Re-organize Documentation website

### PR DESCRIPTION
**Pull request by chibenwa**
Preview me https://rawgit.com/chibenwa/james-project/JAMES-2394-preview/target/site/server/config.html

This change set features:
 - Categorization of configuration page
 - Per product config file listing
 - Missing configurations pages had been added
 - Navigation sections for webadmin, CLI
 - Anti-Spam doc
 - More clear deprecation/experimental mentions
 - And much more....

### CODE REVIEW

**Comment by chibenwa** [comment]
test this please

File: src/site/xdoc/server/config-jmx.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+<body>
+
+  <section name="JMX configuration">
+
+      <subsection name="jmx.properties">
+
+          <p>Consult <a href="https://github.com/apache/james-project/tree/master/server/app/src/main/resources/jmx.properties">jmx.properties</a> in GIT to get some examples and hints.</p>
+
+          <p>This is used to configure the JMX MBean server via which management can be achieved (also used by via the james-cli).
```
Comment (rouazana) [inline src/site/xdoc/server/config-jmx.xml]: s/by via/by/

File: src/site/xdoc/server/config-jmx.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          Alternatives includes <a href="manage-webadmin.html">WebAdmin</a></p>
```
Comment (rouazana) [inline src/site/xdoc/server/config-jmx.xml]: s/includes/include/

File: src/site/xdoc/server/config-james-database.xml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    <properties>
+        <title>Apache James Server 3 - JMX Configuration</title>
+    </properties>
+
+    <body>
+
+        <section name="JPA database configuration">
+
+            <p>This section applies for Spring wiring when JPA backend is chose, as well as Guice JPA's wirings.</p>
```
Comment (rouazana) [inline src/site/xdoc/server/config-james-database.xml]: s/chose/chosen

File: src/site/xdoc/server/config-james-database.xml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            <subsection name="james-database.properties">
+
+                <p>This configuration file is only relevant when using JPA, with Spring or Guice.</p>
+
+                <p>Consult <a href="https://github.com/apache/james-project/tree/master/server/app/src/main/resources/james-database.properties">james.properties</a> in GIT to get some examples and hints.</p>
+
+                <p>The database connection in database.properties</p>
+
+                <p>James has the capacity to use a JDBC-compatible database for storage of both message and user
+                    data. This section explains how to configure James to utilize a database for storage.</p>
```
Comment (rouazana) [inline src/site/xdoc/server/config-james-database.xml]: s/utilize/use?

File: src/site/xdoc/server/config-james-database.xml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                    privileges (select, insert, delete into tables, and on initial startup creation of tables) and
+                    with sufficient quota for the data to be inserted into the database must be available.</p>
+
+                <p>Also, since James will use JDBC to access the database, an appropriate JDBC driver must be
+                    available for installation. You can place the JDBC driver jar in the conf/lib folder, it will
+                    be automatically loaded.</p>
+
+                <dl>
+                    <dt><strong>database.driverClassName</strong></dt>
+                    <dd>he class name of the database driver to be used.</dd>
```
Comment (rouazana) [inline src/site/xdoc/server/config-james-database.xml]: s/he/The

File: src/site/xdoc/server/config-james-database.xml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+                    <dt><strong>database.url</strong></dt>
+                    <dd>The JDBC connection URL for your database/driver.</dd>
+                    <dt><strong>database.username</strong></dt>
+                    <dd>The user id of the database account to be used by this connection.</dd>
+                    <dt><strong>database.password</strong></dt>
+                    <dd>The password of the database account to be used by this connection.</dd>
+                    <dt><strong>vendorAdapter.database</strong></dt>
+                    <dd>Supported adapters are: DB2, DERBY, H2, HSQL, INFORMIX, MYSQL, ORACLE, POSTGRESQL, SQL_SERVER, SYBASE .</dd>
+                    <dt><strong>openjpa.streaming</strong></dt>
+                    <dd>true or false - Use streaming for Blobs. This is only supported on a limited set of databases atm. You
```
Comment (rouazana) [inline src/site/xdoc/server/config-james-database.xml]: s/atm/at the moment/?

File: src/site/xdoc/server/config-james-database.xml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                        should check if its supported by your DB before enable it. See <a href="http://openjpa.apache.org/builds/latest/docs/manual/ref_guide_mapping_jpa.html">http://openjpa.apache.org/builds/latest/docs/manual/ref_guide_mapping_jpa.html</a> (#7.11. LOB Streaming).</dd>
```
Comment (rouazana) [inline src/site/xdoc/server/config-james-database.xml]: s/enable/enabling?

File: src/site/xdoc/server/config-tika.xml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                <dd>IP or domain name of your Tika server. The default value is 127.0.0.1</dd>
+                <dt><strong>tika.port</strong></dt>
+                <dd>Port of your tika server. The default value is 9998</dd>
+                <dt><strong>tika.timeoutInMillis</strong></dt>
+                <dd>Timeout when issuing request to the tika server. The default value is 3 seconds.</dd>
+            </dl>
+
+            Note: You can launch a tika server using this command line:
+
+            <code><pre>docker run --name tika logicalspark/docker-tikaserver:1.15rc2</pre></code>
```
Comment (rouazana) [inline src/site/xdoc/server/config-tika.xml]: maybe adding a warning about slowness (and desactivating some indexes) would be a good thing

File: src/site/xdoc/server/config-webadmin.xml [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+                    <dd>Port the server listens on</dd>
+
+                    <dt><strong>https.enabled</strong></dt>
+                    <dd>true/false. Should HTTPS be enabled? Defaults to false.</dd>
+
+                    <dt><strong>https.keystore</strong></dt>
+                    <dd>Keystore to be used to create secure connections.</dd>
+
+                    <dt><strong>https.password</strong></dt>
+                    <dd>Password to unlock keystore</dd>
```
Comment (rouazana) [inline src/site/xdoc/server/config-webadmin.xml]: Missing final dot.

File: src/site/xdoc/server/config-batchsizes.xml [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+    <properties>
+        <title>Apache James Server 3 - Batch sizes Configuration</title>
+    </properties>
+
+    <body>
+
+        <section name="Batch sizes configuration">
+
+            <p>Limits on operations to be done at the same time. The goal is to avoid to overwhelm the database?</p>
```
Comment (rouazana) [inline src/site/xdoc/server/config-batchsizes.xml]: why a '?' at the end?

File: src/site/xdoc/server/config-batchsizes.xml [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+            <p>This file is for now specific to Cassandra guice wiring.</p>
```
Comment (rouazana) [inline src/site/xdoc/server/config-batchsizes.xml]: s/guice/Guice (and everywhere else)

File: src/site/xdoc/server/config-batchsizes.xml [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+                    <dd>Maximum number of messages to read at the same time at the 'metadata' fetch level.</dd>
+
+                    <dt><strong>fetch.header</strong></dt>
+                    <dd>Maximum number of messages to read at the same time at the 'header' fetch level.</dd>
+
+                    <dt><strong>fetch.body</strong></dt>
+                    <dd>Maximum number of messages to read at the same time at the 'body' fetch level.</dd>
+
+                    <dt><strong>fetch.full</strong></dt>
+                    <dd>Maximum number of messages to read at the same time at the 'full' fetch level.</dd>
```
Comment (rouazana) [inline src/site/xdoc/server/config-batchsizes.xml]: maybe explaining that 'metadata', 'header', 'body' and 'full' are from an RFC could help to understand the meaning of these sentences

File: src/site/xdoc/server/config-antispam.xml [hunk 13]
```diff
@@ -53,7 +58,10 @@
       <li>SpamAssassin Mailet is designed to classify the messages as spam or not 
           with an configurable score threshold. Usually a message will only be 
           considered as spam if it matches multiple criteria; matching just a single test 
-          will not usually be enough to reach the threshold.</li>
+          will not usually be enough to reach the threshold. Note that Spam is reported on a
+          per user bases. As a result of this mailet execution, the <i>org.apache.james.spamassassin.flag</i>
```
Comment (rouazana) [inline src/site/xdoc/server/config-antispam.xml]: s/bases/base

File: src/site/xdoc/server/config-mailbox.xml [hunk 14]
```diff
@@ -37,8 +37,7 @@
 
       <dl>
         <dt><strong>provider</strong></dt>
-        <dd>Supported providers are: jpa (default), jcr, maildir, memory. Be aware that maildir will only work on unix like operation systems!
-        Cassandra mailbox need to be compiled and run using java 8.</dd>
+        <dd>Supported providers are: jpa (default), jcr, maildir, memory. Be aware that maildir will only work on unix like operation systems!</dd>
```
Comment (rouazana) [inline src/site/xdoc/server/config-mailbox.xml]: s/operation/operating

File: src/site/xdoc/server/config-mailbox.xml [hunk 15]
```diff
@@ -37,7 +37,8 @@
 
       <dl>
         <dt><strong>provider</strong></dt>
-        <dd>Supported providers are: jpa (default), jcr, maildir, memory. Be aware that maildir will only work on unix like operation systems!</dd>
+        <dd>Supported providers are: jpa (default), memory.<br/>
+            Experimental providers includes: jcr, hbase, maildir. Be aware that maildir will only work on unix like operation systems!</dd>
```
Comment (rouazana) [inline src/site/xdoc/server/config-mailbox.xml]: s/operation/operating

File: src/site/xdoc/server/monitor-logging.xml [hunk 16]
```diff
@@ -27,7 +27,10 @@
 
   <section name="Monitor via Logging">
   
-    <p>Use the well-known log4j system.</p>
+    <p>Spring wiring uses the well-known log4j system. This page will describe how to configure it. Please note that
+    Guice wiring uses <a href="https://logback.qos.ch/manual/configuration.html">logback</a> as a logging framework,
+    and logging into <a href="https://github.com/internetitem/logback-elasticsearch-appender">ElasticSearch</a> in
+    order to ge Kibana display is supported on top of Guice.</p>
```
Comment (rouazana) [inline src/site/xdoc/server/monitor-logging.xml]: s/ge/get/
Anyway I don't find this sentence very clear. Maybe splitting it in 2 would be better.

**Comment by chibenwa** [comment]
The idea is once we got a better agreement among us, share this to the community to gather their feedbacks.

**Comment by aduprat** [comment]
Build triggered. sha1 is original commit.

**Comment by aduprat** [comment]
Build triggered. sha1 is original commit.

**Comment by aduprat** [comment]
Build started sha1 is original commit.

**Comment by aduprat** [comment]
Build triggered. sha1 is original commit.

**Comment by aduprat** [comment]
Build started sha1 is original commit.

**Comment by aduprat** [comment]
Build triggered. sha1 is original commit.

**Comment by aduprat** [comment]
Build started sha1 is original commit.

**Comment by chibenwa** [comment]
Abort controversial work hanging out for to long.

**Pull request closed without merging**